index.js

```diff
@@ -221,35 +221,45 @@ function createServers (done) {
   ipc.send('listen', opts, function (err, port) {
     if (err) throw new Error('fatal: could not get port', err)
     global.osmServerHost = '127.0.0.1:' + port
-    logger.info(global.osmServerHost)
+    logger.info('Server listening:', global.osmServerHost)
     done()
   })
 }
 
 function notifyReady (done) {
-  win.webContents.on('did-finish-load', () => {
-    setTimeout(() => {
-      var IS_TEST = process.env.NODE_ENV === 'test'
-      if (IS_TEST) win.setSize(1000, 800, false)
-      if (argv.debug) win.webContents.openDevTools()
-
-      win.maximize()
-      splash.destroy()
-      win.show()
-      updater.periodicUpdates()
-      done()
-    }, 1000)
-  })
+  logger.info('Server ready, checking front-end is loaded')
+  // If the window is still loading, wait for it to finish before continuing
+  // win.webContents.isLoading() does not seem to work here on Windows
+  if (!win._didFinishLoad) {
+    logger.info('Front-end still loading, check again once loaded')
+    win.webContents.once('did-finish-load', () => {
+      win._didFinishLoad = true
```

Comment:
This is not critical in this case, but let's not get in the habit of adding properties to objects where the lifecycle is managed by upstream modules (in this case electron window)

Reply:
Yes, very much agreed. We need a separate state management for each process (starting, started, closing, closed, errored). This will serve as a hacky short-term fix.

Reply:
Sure. Something to watch for if we update electron.